table-api/src/main/java/io/deephaven/api/filter/FilterIsNotNull.java

```diff
@@ -3,32 +3,42 @@
  */
 package io.deephaven.api.filter;
 
-import io.deephaven.api.ColumnName;
 import io.deephaven.annotations.SimpleStyle;
+import io.deephaven.api.expression.Expression;
 import org.immutables.value.Value.Immutable;
 import org.immutables.value.Value.Parameter;
 
 /**
- * Evaluates to {@code true} when the given {@link #column() column} value is not {@code null}.
+ * Evaluates to {@code true} when the given {@link #expression() expression} evaluates to not {@code null}.
  */
 @Immutable
 @SimpleStyle
 public abstract class FilterIsNotNull extends FilterBase {
 
-    public static FilterIsNotNull of(ColumnName column) {
-        return ImmutableFilterIsNotNull.of(column);
+    public static FilterIsNotNull of(Expression expression) {
+        return ImmutableFilterIsNotNull.of(expression);
     }
 
     /**
-     * The column name.
+     * The expression.
      *
-     * @return the column name
+     * @return the expression
      */
     @Parameter
-    public abstract ColumnName column();
+    public abstract Expression expression();
+
+    /**
+     * Equivalent to {@code Filter.isNull(expression())}.
+     *
+     * @return the inverse filter
+     */
+    @Override
+    public final FilterIsNull inverse() {
+        return Filter.isNull(expression());
+    }
 
     @Override
-    public final <V extends Visitor> V walk(V visitor) {
+    public final <V extends Filter.Visitor> V walk(V visitor) {
```

Comment:
I think both patterns are supported, and it's more about how the visitor is implemented rather than the interface/name.

I typically like to structure my code so a visitor applies to a single node, and if recursion is needed, it will create a new visitor for each parent it needs to visit.

Reply:
TBD naming to accept, acceptVisitor, dispatch?